wallet-new/src/Cardano/Wallet/Server/LegacyPlugins.hs

```diff
@@ -211,37 +219,18 @@ syncWalletWorker coreConfig = pure $ const $
     modifyLoggerName (const "syncWalletWorker") $
     (view (lensOf @SyncQueue) >>= processSyncRequest coreConfig)
 
-corsMiddleware :: Middleware
-corsMiddleware = cors (const $ Just policy)
-    where
-      policy = simpleCorsResourcePolicy
-        { corsRequestHeaders = ["Content-Type"]
-        , corsMethods = "PUT" : simpleMethods
-        }
+-- | "Attaches" the middlewares to this 'Application'.
+withMiddlewares :: [Middleware] -> Application -> Application
+withMiddlewares = flip $ foldr ($)
 
-
--- | "Attaches" the middleware to this 'Application', if any.  When running in
--- debug mode, chances are we want to at least allow CORS to test the API with
--- a Swagger editor, locally.
-withMiddleware
-    :: MonadIO m
-    => WalletConfiguration
-    -> RunMode
-    -> Application
-    -> m Application
-withMiddleware walletConfiguration wrm app = do
```

Comment:
Middlewares are untangled and defined seperatly which should help with readibility and extending in the future